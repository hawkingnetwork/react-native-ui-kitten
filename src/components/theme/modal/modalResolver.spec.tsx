import React from 'react';
import {
  View,
  Text,
  Button,
} from 'react-native';
import {
  fireEvent,
  render,
  RenderAPI,
  shallow,
} from 'react-native-testing-library';
import { ModalResolver } from './modalResolver.component';

jest.useFakeTimers();

describe('@modal resolver component checks', () => {

  it('* modal resolver component renders properly', () => {
    const modal1: RenderAPI = render(
      <ModalResolver
        visible={true}
        allowBackdrop={false}>
        <View>
          <Text>
            Test1
          </Text>
        </View>
      </ModalResolver>,
    );

    const modal2: RenderAPI = render(
      <ModalResolver
        visible={false}
        allowBackdrop={false}>
        <View>
          <Text>
            Test2
          </Text>
        </View>
      </ModalResolver>,
    );

    const { output: firstOutput } = shallow(modal1.getByType(ModalResolver));
    const { output: secondOutput } = shallow(modal2.getByType(ModalResolver));

    expect(firstOutput).toMatchSnapshot();
    expect(secondOutput).toMatchSnapshot();
  });

  it('* modal resolver component props checks', () => {
    const modalPassingProps = {
      visible: true,
      allowBackdrop: false,
    };
    const modal = <ModalResolver {...modalPassingProps}/>;

    expect(modal.props.visible).toBe(modalPassingProps.visible);
    expect(modal.props.allowBackdrop).toBe(modalPassingProps.allowBackdrop);
  });

  it('* modal resolver backdrop press calling checks', () => {
    const onBackdropPress = jest.fn();

    const component: RenderAPI = render(
      <ModalResolver
        visible={true}
        allowBackdrop={true}>
        <View>
          <Text>Test1</Text>
          <Button
            title='Close Modal'
            onPress={onBackdropPress}
          />
        </View>
      </ModalResolver>,
    );

    fireEvent.press(component.getByType(Button));
    expect(onBackdropPress).toHaveBeenCalled();
  });

});
