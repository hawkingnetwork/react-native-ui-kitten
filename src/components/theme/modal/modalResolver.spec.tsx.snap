// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modal resolver component checks * modal resolver component renders properly 1`] = `
<View
  style={
    Object {
      "height": 0,
      "position": "absolute",
      "width": 0,
    }
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "alignSelf": "flex-start",
      }
    }
    visible={true}
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <Text>
        Test1
      </Text>
    </View>
  </View>
</View>
`;

exports[`@modal resolver component checks * modal resolver component renders properly 2`] = `null`;
